src/main/java/org/jenkinsci/main/modules/instance_identity/InstanceIdentity.java

```diff
@@ -53,19 +53,32 @@ public InstanceIdentity(File keyFile) throws IOException {
             keys = read(new FileReader(oldKeyFile), gen);
             write(keys, keyFile);
             Util.deleteFile(oldKeyFile);
-        } else if (keyFile.exists()) {
-            byte[] enc = FileUtils.readFileToByteArray(keyFile);
+        } else {
+            Reader reader = getFileReader(keyFile);
+            if (reader!=null && keyFile.exists()) {
+                keys = read(reader, gen);
```

Comment:
at this point if `keyFile.exists()` is false `reader` is `null`. So checking if `reader!=null` should suffice?